src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -121,6 +121,16 @@ std::vector<image_annotation> parse_annotations(
         for (const auto& coord_kv : coordinates) {
 
           const flex_string& coord_key = coord_kv.first.get<flex_string>();
+
+          // Check the type of coordinate
+          if (coord_kv.second.get_type() != flex_type_enum::FLOAT and
+              coord_kv.second.get_type() != flex_type_enum::INTEGER) {
+            log_and_throw(
+                "Invalid object annotations discovered. Annotation coordinate "
+                "'" +
+                coord_key + "' must have type of Int or Float.");
```

Comment:
"Bounding box co-ordinates must be int or float values" is more accurate.

Reply:
good!